workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -76,91 +75,96 @@ private Worker(string workerType, Connection connection, ILogDispatcher logDispa
+        public static async Task<Worker> CreateWorkerAsync(
+            ReceptionistConfig config, 
+            ConnectionParameters connectionParameters,
+            ILogDispatcher logger, Vector3 origin)
         {
-            using (var locator = new Locator(config.LocatorHost, config.LocatorParameters))
+            using (var connectionFuture =
+                Connection.ConnectAsync(config.ReceptionistHost, config.ReceptionistPort, config.WorkerId, connectionParameters))
             {
-                var deploymentList = await GetDeploymentList(locator);
+                return await TryToConnect(connectionFuture, connectionParameters.WorkerType, logger, origin);
+            }
+        }
 
-                var deploymentName = deploymentListCallback(deploymentList);
-                if (String.IsNullOrEmpty(deploymentName))
+        public static async Task<Worker> CreateWorkerAsync(
+            LocatorConfig parameters,
+            ConnectionParameters connectionParameters,
+            ILogDispatcher logger, Vector3 origin)
+        {
+                using (var locator = new Locator(parameters.LocatorHost, parameters.LocatorParameters))
                 {
-                    throw new ConnectionFailedException("No deployment name chosen",
-                        ConnectionErrorReason.DeploymentNotFound);
-                }
+                    var deploymentList = await GetDeploymentList(locator);
 
-                var connectionParams = config.CreateConnectionParameters();
-                using (var connectionFuture = locator.ConnectAsync(deploymentName, connectionParams, (_) => true))
-                {
-                    var connection = await Task.Run(() => connectionFuture.Get());
-                    if (!connection.IsConnected)
+                    var deploymentName = parameters.DeploymentListCallback(deploymentList);
+                    if (string.IsNullOrEmpty(deploymentName))
                     {
-                        throw new ConnectionFailedException(GetConnectionFailureReason(connection),
-                            ConnectionErrorReason.CannotEstablishConnection);
+                        throw new ConnectionFailedException("No deployment name chosen",
+                            ConnectionErrorReason.DeploymentNotFound);
                     }
 
-                    // A check is needed for the case that play mode is exited before the connection can complete.
-                    if (!Application.isPlaying)
+                    using (var connectionFuture = locator.ConnectAsync(deploymentName, connectionParameters, (_) => true))
                     {
-                        connection.Dispose();
-                        throw new ConnectionFailedException("Editor application stopped",
-                            ConnectionErrorReason.EditorApplicationStopped);
+                        var connection = await Task.Run(() => connectionFuture.Get());
```

Comment:
Can we not use the `TryToConnect` function here?

Reply:
yes we can